src/stdio_mgr/stdio_mgr.py

```diff
@@ -270,34 +274,56 @@ def stdio_mgr(in_str="", close=True):
         initially empty.
 
     """
-    if close:
-        out_cls = SafeCloseRandomTextIO
-        in_cls = SafeCloseTeeStdin
-    else:
-        out_cls = RandomTextIO
-        in_cls = TeeStdin
-
-    old_stdin = sys.stdin
-    old_stdout = sys.stdout
-    old_stderr = sys.stderr
-
-    with ExitStack() as stack:
-        new_stdout = stack.enter_context(out_cls())
-        new_stderr = stack.enter_context(out_cls())
-        new_stdin = stack.enter_context(in_cls(new_stdout, in_str))
-
-        close_files = stack.pop_all().close
-
-    sys.stdin = new_stdin
-    sys.stdout = new_stdout
-    sys.stderr = new_stderr
-
-    try:
-        yield new_stdin, new_stdout, new_stderr
-    finally:
-        sys.stdin = old_stdin
-        sys.stdout = old_stdout
-        sys.stderr = old_stderr
 
+    def __new__(cls, in_str="", close=True):
+        """Instantiate new context manager that is also a tuple."""
         if close:
-            close_files()
+            out_cls = SafeCloseRandomTextIO
+            in_cls = SafeCloseTeeStdin
+        else:
+            out_cls = RandomTextIO
+            in_cls = TeeStdin
+
+        stdout = out_cls()
+        stderr = out_cls()
+        stdin = in_cls(stdout, in_str)
+
+        self = super(stdio_mgr, cls).__new__(cls, [stdin, stdout, stderr])
+
+        self.stdin = stdin
+        self.stdout = stdout
+        self.stderr = stderr
+        self._close = close
+
+        return self
+
+    def __enter__(self):
+        """Enter context, replacing sys stdio objects with capturing streams."""
+        self.old_stdin = sys.stdin
+        self.old_stdout = sys.stdout
+        self.old_stderr = sys.stderr
+
+        with ExitStack() as stack:
+            stack.enter_context(self[0])
```

Comment:
`self[0]` could be `self.stdin` , etc

likewise below

Reply:
Definitely change these, once the `@property`s are implemented!

Reply:
Went a slightly different route, creating super class `_MultiCloseContextManager`.